api/src/services/subscription-service.ts

```diff
@@ -82,58 +82,68 @@ export class SubscriptionService {
 		await this.subscriptionPool.add(channelAddress, res.subscriber, subscriberId, false);
 		await this.addSubscription(subscription);
 		await this.channelInfoService.addChannelSubscriberId(channelAddress, subscriberId);
-		await this.channelInfoService.updateLatestChannelLink(channelAddress, res.subscriptionLink);
 
 		return { seed: res.seed, subscriptionLink: res.subscriptionLink };
 	}
 
 	async authorizeSubscription(
 		channelAddress: string,
-		subscriptionLink: string,
-		publicKey: string,
+		subscription: Subscription,
 		authorSub: Subscription
 	): Promise<AuthorizeSubscriptionBodyResponse> {
 		const lockKey = channelAddress + authorSub.identityId;
 
 		return this.lock.acquire(lockKey).then(async (release) => {
 			try {
+				const { publicKey, subscriptionLink, identityId } = subscription;
 				const presharedKey = authorSub.presharedKey;
 				const streamsAuthor = (await this.subscriptionPool.get(channelAddress, authorSub.identityId, true)) as Author;
+
 				if (!streamsAuthor) {
 					throw new Error(`no author found with channelAddress: ${channelAddress} and identityId: ${authorSub?.identityId}`);
 				}
+
+				const authorPubKey = streamsAuthor.clone().get_public_key();
 				const subscriptions = await subscriptionDb.getSubscriptions(channelAddress);
-				const existingSubscriptions = subscriptions.filter((s) => s.type === SubscriptionType.Subscriber && s.isAuthorized === true && s.publicKey);
-				const existingSubscriptionKeys = existingSubscriptions.map((s) => s.publicKey);
+				const existingSubscriptions = subscriptions.filter(
```

Comment:
to prevent exceptions I would add optional chaining for `subscriptions?.filter`, `s?.isAuthorized`, `s?.accessRights`, and `existingSubscriptions?.map()?.filter()`

Reply:
Thanks, I've added a check for empty subscriptions. If I just use ?. I would receive and error when destructuring:
`[..., ...existingSubscriptionKeys]`